File: app/components/IndentedLabel.js
import {Label} from '@nativescript/core/ui/label';

const paint = new android.graphics.Paint();
paint.setStyle(android.graphics.Paint.Style.STROKE);
paint.setColor(android.graphics.Color.parseColor('#282828'));
paint.setStrokeWidth(5);

const factor = 60;

export default class IndentedLabel extends Label {
  createNativeView() {
    const TV = android.widget.TextView.extend({
      depth: 0,
      paddingBottom: 10,
      onDraw(canvas) {
        this.super.setPadding(factor * this.depth + 20, 0, 10, this.paddingBottom);
        this.super.onDraw(canvas);
        for (let i = 1; i <= this.depth; i++) {
          const indent = i * factor;
          canvas.drawLine(indent, 0, indent, this.super.getHeight(), paint);
        }
      },
      setDepth(depth, paddingBottom) {
        this.depth = depth;
        this.paddingBottom = paddingBottom;
      },
    });
    return new TV(this._context);
  }
}

File: app/src/main/java/de/max/roehrl/vueddit2/ui/view/IndentedLabel.kt
package de.max.roehrl.vueddit2.ui.view

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.content.ContextCompat
import de.max.roehrl.vueddit2.R

class IndentedLabel(context: Context, attrs: AttributeSet) : AppCompatTextView(context, attrs) {
    private var depth = 0
    private val factor = 40
    private var paddingTop2 = 10
    private var paddingBottom2 = 10
    private val paint = Paint().apply {
        style = Paint.Style.FILL
        color = ContextCompat.getColor(context, R.color.comment_indent_lines)
        strokeWidth = 0f
    }

    override fun onDraw(canvas: Canvas?) {
        setPadding(factor * depth + 20, paddingTop2, 10, paddingBottom2)
        for (i in 0..depth) {
            val indent = i * factor
            canvas?.drawRect(indent + 0f, 0f, indent + 5f, height.toFloat(), paint)
        }
        super.onDraw(canvas)
    }

    fun setDepth(depth: Int, paddingTop: Int = 10, paddingBottom: Int = 10) {
        this.depth = depth
        paddingTop2 = paddingTop
        paddingBottom2 = paddingBottom
    }
}